client/sites/tools/logs/index.tsx

```diff
@@ -251,3 +263,239 @@ export const SiteLogs = ( {
+	} );
+
+	return {
+		data: data?.logs ? data.logs : EMPTY_ARRAY,
+		paginationInfo: {
+			totalItems: 0, // TODO
+			totalPages: 1, // TODO
+		},
+		isLoading: isFetching,
+	};
+};
+
+const useFields = ( { logType }: { logType: LogType } ) => {
+	if ( logType === 'php' ) {
+		return [
+			{ id: 'severity' },
+			{ id: 'timestamp' },
+			{ id: 'message' },
+			{ id: 'kind' },
+			{ id: 'name' },
+			{ id: 'file' },
+			{ id: 'line' },
+		];
+	}
+
+	return [
+		{ id: 'request_type' },
+		{ id: 'timestamp' },
+		{ id: 'status' },
+		{ id: 'request_url' },
+		{ id: 'body_bytes_sent' },
+		{ id: 'cached' },
+		{ id: 'http_host' },
+		{ id: 'http_referer' },
+	];
+};
+
+const getVisibleFieldsForLogType = ( logType: LogType ) => {
+	if ( logType === 'php' ) {
+		return [ 'severity', 'timestamp', 'message' ];
+	}
+	return [ 'request_type', 'timestamp', 'status', 'request_url' ];
+};
+
+export const SiteLogsDataViews = ( { logType }: { logType: LogType } ) => {
+	// TODO:
+	// - Empty states: "no entries in this time range" instead of DataViews's default.
+	// - DataViews: filter, sort, pagination.
+	// - DataViews: max width for long cells.
+	// - DataViews: address the "show more" interaction.
+	// - DataViews: spacing left/right.
+
+	const { __ } = useI18n(); // Can we use translate instead?
+
+	const moment = useLocalizedMoment();
+	const getLatestDateRange = useCallback( () => {
+		const startTime = moment().subtract( 7, 'd' );
+		const endTime = moment();
+		return { startTime, endTime };
+	}, [ moment ] );
+
+	const [ dateRange, setDateRange ] = useState( () => {
+		const latest = getLatestDateRange();
+		const dateRangeQuery = getDateRangeQueryParam( moment );
+		return {
+			startTime: dateRangeQuery.startTime || latest.startTime,
+			endTime: dateRangeQuery.endTime || latest.endTime,
+		};
+	} );
+	const handleTimeRangeChange = ( newStart: Moment | null, newEnd: Moment | null ) => {
+		if (
+			( ! newStart && ! newEnd ) ||
+			( dateRange.startTime.isSame( newStart ) && dateRange.endTime.isSame( newEnd ) )
+		) {
+			return;
+		}
+
+		// setIsMobileOpen( false ); // TODO
+		let startTime = newStart || dateRange.startTime;
+		let endTime = newEnd || dateRange.endTime;
+		if ( ! startTime.isValid() || ! endTime.isValid() ) {
+			const latest = getLatestDateRange();
+			startTime = latest.startTime;
+			endTime = latest.endTime;
+		}
+
+		setDateRange( { startTime, endTime } );
+		// setAutoRefresh( false ); // TODO
+		updateDateRangeQueryParam( { startTime, endTime } );
+	};
+
+	const [ view, setView ] = useState( () => {
+		return {
+			type: 'table' as const,
+			perPage: 10, // TODO
+			page: 1, // TODO
+			fields: getVisibleFieldsForLogType( logType ),
+		};
+	} );
+	const fields = useFields( { logType } );
+	const { data, paginationInfo, isLoading } = useDataLogs( { view, logType, dateRange } );
+	const onChangeView = ( newView: View ) =>
+		setView( ( oldView ) => ( {
+			...oldView,
+			...newView,
+			type: 'table' as const,
```

Comment:
Do we need this?

Reply:
Yes, if you don't add `as const`, TS will view `type` as a mutable `string` and dataviews expect strictly `table`, `grid` and two other values I can't recall.